commit fdb598690120ba314f3621a9d3fea4e6f0669ccf (refs/changes/55/5655/1)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2019-02-28T01:11:23+00:00 (7 months ago)
    
    MANTA-4137 comment about garbage-collector 'cleaned_state' should be clearer

diff --git a/lib/delete_record_transformer.js b/lib/delete_record_transformer.js
index 379b9cb..7e9187d 100644
--- a/lib/delete_record_transformer.js
+++ b/lib/delete_record_transformer.js
@@ -170,13 +170,15 @@ _process_record(record, done)
 	}
 
 	/*
-	 * In the MorayDeleteRecordCleaner, Moray#batch may fail a potentially
-	 * large group of delete operations if one of those operations is to
-	 * delete a key that doesn't exist in the delete queue. For this reason,
-	 * we must make sure that no component of the garbage-collector attempts
-	 * requests deletion of the same key twice. This shared status object
-	 * determines whether the MakoInstructionUploader has already requested
-	 * that a record for some key be removed.
+	 * The DeleteRecordTransformer emits a separate instruction for each
+	 * storage node the object referred to by the record's key is stored on.
+	 * Since the metadata record occupies a single row in the metadata tier,
+	 * we use this bit of state to ensure that the key is included in only
+	 * one batch delete.
+	 *
+	 * A Moray batch delete could fail a potentially large number of
+	 * requests if any given delete transaction fails, so not doing this
+	 * opens the collector up to potentially wasting a lot of work.
 	 */
 	var cleaned_state = {
 		cleaned: false
